disnake/embeds.py

```diff
@@ -559,18 +594,40 @@ def author(self) -> _EmbedAuthorProxy:
         """
         return cast("_EmbedAuthorProxy", EmbedProxy(self._author))
 
+    @overload
+    def set_author(
+        self, *, name: Any, url: Optional[Any] = ..., icon_url: Optional[Any] = ...
+    ) -> Self:
+        ...
+
+    @overload
+    def set_author(self, *, name: Any, url: Optional[Any] = ..., icon_file: File = ...) -> Self:
+        ...
+
     def set_author(
         self,
         *,
         name: Any,
         url: Optional[Any] = None,
-        icon_url: Optional[Any] = None,
+        icon_url: Optional[Any] = MISSING,
+        icon_file: File = MISSING,
     ) -> Self:
         """Sets the author for the embed content.
 
         This function returns the class instance to allow for fluent-style
         chaining.
 
+        At most one of ``icon_url`` or ``icon_file`` may be passed.
+
+        .. warning::
+            Passing a :class:`disnake.File` object will make the embed not
+            reusable.
+
+        .. warning::
+            If used with the other ``set_*`` methods you must ensure
+            that the :attr:`.File.filename` is different than the other file(s)
+            that you are passing.
```

Comment:
```suggestion
            If used with the other ``set_*`` methods, you must ensure
            that the :attr:`.File.filename` is unique to avoid duplication.
```